### PR TITLE
JAPI-127 Add FileSpray/DownloadFile to JAPI

### PR DESCRIPTION
**Pull request by jpmcmu**
- Added downloadFile API
- Added downloadFile test

Signed-off-by: James McMullan <James.McMullan@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [ ] My code follows the code style of this project.
  - [x] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [x] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: wsclient/src/test/java/org/hpccsystems/ws/client/platform/test/PlatformTester.java [hunk 1]
```diff
@@ -269,6 +276,41 @@ else if (currentParam.matches(WSSQLPORTPATTERN))
...
 
+            // Test file download
+            System.out.println("Download test ...");
+            String fileName = null;
+            for (int i = 0; pfs != null && i < pfs.length; i++) 
+            {
+                if (pfs[i].getIsDir() == false
+                && pfs[i].getFilesize() < 4 * 1024 * 1024)  // Only download small files for the test
+                {
+                    fileName = pfs[i].getName();
```
Comment (rpastrana) [inline wsclient/src/test/java/org/hpccsystems/ws/client/platform/test/PlatformTester.java]: should we break after the first small file?

**Comment by jpmcmu** [comment]
@rpastrana Implemented code review changes & squashed

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java [hunk 2]
```diff
@@ -938,6 +941,125 @@ private boolean uploadFile(File file, DropZone dropZone) throws Exception
...
+
+        return bytesTransferred;
+    }
+
+    /**
+     * Attempts to download the specified file from a dropzone
+     *
+     * @param outputChannel 
+     *            - The FileChannel to write to
+     * @param dropZones
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java]: this one should change as well.

**Pull request merged**